src/Core/Form/IdentifiableObject/DataProvider/ProductFormDataProvider.php

```diff
@@ -233,4 +245,42 @@ private function presentTags(array $localizedTagsList): array
 
         return $tags;
     }
+
+    /**
+     * @param ProductForEditing $productForEditing
+     *
+     * @return array
+     */
+    private function extractSuppliersData(ProductForEditing $productForEditing): array
+    {
+        /** @var ProductSupplierOptions $productSupplierOptions */
+        $productSupplierOptions = $this->queryBus->handle(new GetProductSupplierOptions($productForEditing->getProductId()));
+        $defaultSupplierId = $productSupplierOptions->getDefaultSupplierId();
+
+        $suppliersData = [
+            'default_supplier_id' => $defaultSupplierId,
+        ];
+
+        //@todo: one layer of array is not needed anymore if product has no combinations. Handle it in separate pr
+        foreach ($productSupplierOptions->getOptionsBySupplier() as $index => $supplierOption) {
+            $supplierId = $supplierOption->getSupplierId();
+            $suppliersData['supplier_ids'][$supplierId] = $supplierId;
+            $suppliersData['supplier_references'][$supplierId]['is_default'] = $supplierId === $defaultSupplierId;
+            $suppliersData['supplier_references'][$supplierId]['supplier_id'] = $supplierId;
+            $suppliersData['supplier_references'][$supplierId]['supplier_name'] = $supplierOption->getSupplierName();
+
+            foreach ($supplierOption->getProductSuppliersForEditing() as $supplierForEditing) {
```

Comment:
Since there is only one reference used by the form maybe we should loop through the whole list but only use the first element no? Maybe it would even be worth modifying the query handler result or have two commands one Dedicated to Simple product and another for combinations product?

But anyway there's a problem with this loop it actually replaces the `$suppliersData['supplier_references'][$supplierId]['product_supplier']` part in each iteration which actually results in returning the data from last combination